### PR TITLE
Convert old metrics to torchmetrics in load functions

### PR DESCRIPTION
**Pull request by KnathanM**
## Description
#1166 showed that models trained in v 2.0.4 couldn't be loaded in 2.1.1 because of the metrics not having `_defaults`. 

The metrics list that gets given to `MPNN` when creating a model is added to the model's `hparams` `AttributeDict`. This `AttributeDict` is pickled as is in a model file to be used to recreate the model architecture when loading the model. In chemprop<2.1, our metrics were not torchmetrics, but (after using the v2.0 -> v2.1 conversion script) when we load a model, the metrics will get unpickled using the new torchmetrics classes. This correctly handles things like `task_weights` that are in both class definitions, but torchmetrics classes have more attributes than our original metrics, including `_defaults`. 

When I did the metric overhaul PR #1022, I knew this was a problem for the loss function saved within the predictor's hparams. I included this conversion code in the model loading functions
https://github.com/chemprop/chemprop/blob/1b8f4133b092fb0241f690f93a2b11e7db07c6dd/chemprop/models/model.py#L269-L272

Now I am confused why I didn't do the same thing for the metrics. I have a vague feeling that I didn't do it for the metrics on purpose, but I can't remember what the reason was. In any event, I re-created the error in the linked issue by training a model in 2.0.4 with AUROC metric, converting it using the script, and then trying to load it in v2.1.1, so it looks like something that needs to be fixed. 

## Other notes
I used `Factory` because a couple of the metrics/loss function have extra init args and this seemed like the way to get all the init args that are needed. 

I added two checkpoint files that have most all of the metrics in them and a test to see if they load. This should hopefully catch if something prevents models with defined metrics from being loaded.

### CODE REVIEW

**Comment by KnathanM** [comment]
Unfortunately the task weights aren't actually directly visible in the `__dict__`. They are hidden under `_buffers`. Here is an example for a DirichletLoss object
```
{'training': True,
 '_parameters': OrderedDict(),
 '_buffers': OrderedDict([('task_weights', tensor([[1., 1., 1., 1.]]))]),
 '_non_persistent_buffers_set': set(),
 '_backward_pre_hooks': OrderedDict(),
...
 '_cache': None,
 'total_loss': tensor(0.),
 'num_samples': tensor(0),
 'v_kl': 0.2}
```
So I need to access task weights from the metric's attribute.
